finance_research/extract/extractor.py

```
@@ -0,0 +1,265 @@
+            client = OpenAI(api_key=self.api_key, base_url=self.base_url)
+            schema_hint = (
+                '{"company_name": str, "report_type": str, "fiscal_period": str, '
+                '"revenue": float, "gross_profit": float, "net_profit": float, '
+                '"gross_margin": float, "net_margin": float, '
+                '"operating_cash_flow": float, '
+                '"segment_revenue": [{"segment": str, "amount": float}], '
+                '"region_revenue": [{"segment": str, "amount": float}], '
+                '"evidence": str}'
+            )
+            prompt = (
+                "从以下财报文本中抽取关键财务指标，严格返回 JSON，缺失值用 null。\n"
```

Ruff (0.16.1)

[warning] String contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF001)